app/components/network/networkController.js

```diff
@@ -19,12 +19,82 @@ function ($scope, $state, $stateParams, Network, Messages) {
     });
   };
 
-  $('#loadingViewSpinner').show();
-  Network.get({id: $stateParams.id}, function (d) {
-    $scope.network = d;
-    $('#loadingViewSpinner').hide();
-  }, function (e) {
-    $('#loadingViewSpinner').hide();
-    Messages.error("Failure", e, "Unable to retrieve network info");
+  $scope.containerLeaveNetwork = function containerLeaveNetwork(network, containerId) {
+    $('#loadingViewSpinner').show();
+    Network.disconnect({id: $stateParams.id}, { Container: containerId, Force: false }, function (d) {
+      if (d.message) {
+        $('#loadingViewSpinner').hide();
+        Messages.send("Error", {}, d.message);
+      } else {
+        $('#loadingViewSpinner').hide();
+        Messages.send("Container left network", $stateParams.id);
+        $state.go('network', {id: network.Id}, {reload: true});
+      }
+    }, function (e) {
+      $('#loadingViewSpinner').hide();
+      Messages.error("Failure", e, "Unable to disconnect container from network");
+    });
+  };
+
+  function getNetwork() {
+    $('#loadingViewSpinner').show();
+    Network.get({id: $stateParams.id}, function (d) {
+      $scope.network = d;
+      getContainersInNetwork(d);
+      $('#loadingViewSpinner').hide();
+    }, function (e) {
+      $('#loadingViewSpinner').hide();
+      Messages.error("Failure", e, "Unable to retrieve network info");
+    });
+  };
+
+  function getContainersInNetwork(network) {
+    if (network.Containers) {
+      Container.query({
+        filters: {network: [$stateParams.id]}
+      }, function (containersInNetworkResult) {
+        if ($scope.containersToHideLabels) {
+          containersInNetworkResult = ContainerHelper.hideContainers(containersInNetworkResult, $scope.containersToHideLabels);
+        }
+        var containersInNetwork = [];
+        containersInNetworkResult.forEach(function(container) {
```

Comment:
> The containers in network from the network response is different as it is from the containers API endpoint. I can make the requested change, but that would mean we would have to filter out the information about a container for that specific network (e.g. IP address of that container in the network).

The content of the `network.Containers` will not be sufficient enough for the features related to this PR (you need the container ID to jump to to the containers details view and to trigger the leave network action, the ID is not included in this container object present in this array). 

> The alternative is to iterate through container. NetworkSettings.Networks to find the desired network and use the information from that. I feel that is overkill, since we already have the desired information and we only use the query for containers to fetch the labels and filter out the containers we want to hide.

I do think it's an overkill too and agree with the fact that we should get the data from the `network.Containers` array, but you also need to add the container ID in that structure:

```javascript
        containersInNetworkResult.forEach(function(container) {
          var containerInNetwork = network.Containers[container.Id];
          containerInNetwork.Id = container.Id;
          containersInNetwork.push(containerInNetwork);
        });
```